pydantic_graph/pydantic_graph/mermaid.py

```diff
@@ -56,6 +71,11 @@ def generate_code(  # noqa: C901
     if title:
         lines = ['---', f'title: {title}', '---']
     lines.append('stateDiagram-v2')
+    if direction is not None:
+        valid_directions = {'TB', 'LR', 'RL', 'BT'}
+        if direction not in valid_directions:
+            raise UserError(f"Invalid direction '{direction}'. Must be one of {valid_directions}.")
```

Comment:
```suggestion
```
I don't think we need to check this at runtime, if the user really want to give another value, they can break the diagram, that's fine with us.

Reply:
lol 😂 I love your sense of humor.